rend3/src/managers/mesh.rs

```diff
@@ -442,12 +437,11 @@ impl MeshManager {
 
             mesh.vertex_range = new_vert_range;
             mesh.index_range = new_index_range;
-
-            for &object in &mesh.objects {
-                object_manager.set_mesh_ranges(object, mesh.vertex_range.clone(), mesh.index_range.clone())
-            }
         }
 
+        // Need to call this to update the vertex ranges cached inside the objects
+        object_manager.fix_objects_after_realloc(self, skeleton_manager);
+
```

Comment:
The old method was not taking into account that some of these objects may be animated skeletons, so the vertex range is not always coming from the mesh. I think it's better to just translate this responsibility onto the object manager, who can do cleanup of all the objects in a single pass at the end.

This also gets rid of the list of `objects` references inside the `InternalMesh`, which I think is a great simplification (the less redundancy we have, the better). The same reasoning did not apply to the skeleton manager, so I left the skeleton bookkeeping code as-is.

Reply:
Seems good to me.